src/pages/guides/general/how-to-diagnose-solana-transaction-errors.md

```diff
@@ -0,0 +1,162 @@
+
+- All accounts used
+- All instructions submitted
+- All logs including error messages
+
+This failed transaction can then be sent to someone to inspect the details of the transaction to help diagnose why your transaction is failing.
+
+This works on both **Mainnet** and **Devnet**. This does also work on **Localnet** but is more complicated and sharing the details is more difficult.
+
+### umi
+
+Metaplex Umi's `skipPreflight` can be found in the `sendAndConfirm()` and `send()` function args and can be enabled like so:
+
+#### sendAndConfirm()
+```ts
+const tx = createV1(umi, {
+    ...args
+}).sendAndConfirm(umi, {send: { skipPreflight: true}})
+
+// Convert signature to string
+const signature = base58.deserialize(tx.signature);
+
+// Log transaction signature
+console.log(signature)
+```
+
+#### send()
+```ts
+const tx = createV1(umi, {
+    ...args
+}).send(umi, {skipPreflight: true})
+
+// Convert signature to string
+const signature = base58.deserialize(tx);
+
+// Log transaction signature
+console.log(signature)
+```
+
+### web3js
+
+```ts
+// Create Connection
+const connection = new Connection("https://api.devnet.solana.com", "confirmed",);
+
+// Create your transaction
+const transaction = new VersionedTransaction()
+
+// Add skipPreflight to the sendTransaction() function
+const res = await connection.sendTransaction(transaction, [...signers], {skipPreflight: true})
+
+// Log out the transaction signature
+console.log(res)
+```
+
+### solana-client (rust)
+
+```rust
+// Create Connection
+let rpc_client = rpc_client::RpcClient::new("https://api.devnet.solana.com".to_string());
+
+// Create your transaction
+let transaction = new Transaction()
+
+// Add skipPreflight to the sendTransaction() function
+let res = rpc_client
+    .send_transaction_with_config(&create_asset_tx, RpcSendTransactionConfig {
+        skip_preflight: true,
+        preflight_commitment: Some(CommitmentConfig::confirmed().commitment),
+        encoding: None,
+        max_retries: None,
+        min_context_slot: None,
+    })
+    .await
+    .unwrap();
+
+// Log out the transaction signature
+println!("Signature: {:?}", res)
+```
+
+By logging out the transaction ID you can visit a Solana blockchain explorer and search for the transaction ID which will display the failed transaction.
+
+- SolanaFM
+- Solscan
+- Solana Explorer
+
+This transaction ID or explorer link can the be shared with someone who may be able to assist you.
+
+## Common Types of Errors
+
+There are some common errors that normally occur 
+
+
+### Error Codes xx (23)
+
+While normally complimented with some additional text to describe the error codes these codes can sometimes appear on their own in a non descriptive manor. If this happens and you know the program that threw the error you can sometimes find the program in Github and it will have an errors.rs page that lists out all the possible errors of the program.
```

Comment:
```suggestion
While normally complimented with some additional text to describe the error codes these codes can sometimes appear on their own in a non descriptive manner. If this happens and you know the program that threw the error you can sometimes find the program in Github and it will have an errors.rs page that lists out all the possible errors of the program.
```